### PR TITLE
Improve NFT utils

### PR DESCRIPTION
**Pull request by h0ngcha0**
Mostly rename datatypes and function names to better match the current NFT implementation.

### CODE REVIEW

File: packages/cli/scripts/pre-designed-nft.ts [hunk 1]
```diff
@@ -170,7 +170,7 @@ function validateAttributesTypeIfExists(parent: string, attributes?: object) {
       }
 
       Object.keys(item).forEach((key) => {
-        if (!validNFTMetadataAttributeTypes.includes(typeof item[key])) {
+        if (!validNFTMetadataUriAttributeTypes.includes(typeof item[key])) {
```
Comment (Lbqds) [inline packages/cli/scripts/pre-designed-nft.ts:173]: It appears that this variable does not exist, and after adding `scripts/**/*.ts` [here](https://github.com/alephium/alephium-web3/blob/a8803533be3f08945838679f1bc885139ba98784/packages/cli/tsconfig.json#L18), there are some other errors during the compilation of `pre-designed-nft.ts`.

File: packages/cli/scripts/pre-designed-nft.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ import fs from 'fs'
 import path from 'path'
 import { Configuration, CreateImageRequestSizeEnum, OpenAIApi } from 'openai'
 import { create as ipfsHttpClient } from 'ipfs-http-client'
-import { validateEnumerableNFTBaseUri, NFTMetadata, validNFTMetadataAttributeTypes } from '@alephium/web3'
+import { validateNFTBaseUri, NFTMetadata, validNFTUriMetadataAttributeTypes } from '@alephium/web3'
```
Comment (Lbqds) [inline packages/cli/scripts/pre-designed-nft.ts:22]: `NFTMetadata` => `NFTTokenUriMetaData`

Comment (h0ngcha0) [reply packages/cli/scripts/pre-designed-nft.ts]: :+1:

**Review by Lbqds — APPROVED** [review]
LGTM 👍

**Review by polarker — APPROVED** [review]
LGTM

**Pull request merged**